packages/backend/src/core/reports/charts.ts

```diff
@@ -0,0 +1,26 @@
+import { ChartPoint, UnixTime } from '@l2beat/common'
+
+export function getDailyTimestamps(min: UnixTime, max: UnixTime) {
+  const timestamps: UnixTime[] = []
+  for (let t = min; t.lte(max); t = t.add(1, 'days')) {
+    timestamps.push(t)
+  }
+  return timestamps
+}
+
+export function addMissingDailyTimestamps(points: ChartPoint[]): ChartPoint[] {
+  if (points.length === 0) return []
+  const [min] = points[0]
+  const [max] = points[points.length - 1]
+  const daily = getDailyTimestamps(min, max)
+
+  return daily.reduce((acc, timestamp, i) => {
```

Comment:
This algorithm would be more efficient and cleaner if the reduce was just replaced with the loop from getDailyTimestamps and getDailyTimestamps would go away.